core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -284,26 +284,34 @@ class BrokerTopicMetrics(name: Option[String], remoteStorageEnabled: Boolean = f
   }
 
   case class GaugeWrapper(metricType: String) {
-    @volatile private var gaugeObject: Gauge[Long] = _
-    final private val gaugeLock = new Object
-    final val aggregatedMetric = new AggregatedMetric()
+    // The map to store:
+    //   - per-partition value for topic-level metrics. The key will be the partition number
+    //   - per-topic value for broker-level metrics. The key will be the topic name
+    private val metricValues = new ConcurrentHashMap[String, Long]()
+    private val meterLock = new Object
```

Comment:
could you rename it back to `gaugeLock`?

Reply:
Updated it. Thank you.